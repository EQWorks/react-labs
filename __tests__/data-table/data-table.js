import React from 'react'
import renderer from 'react-test-renderer'
import DataTable from '../../src/deprecated/data-table'


beforeEach(() => {
  jest.spyOn(console, 'error')
  jest.spyOn(console, 'warn')
})

afterEach(() => {
  /* eslint-disable no-console */
  expect(console.error).not.toBeCalled()
  expect(console.warn).not.toBeCalled()
})

describe('<DataTable/>', () => {
  it('should render successfully', async () => {
    let data = {
      data: [],
      isPercentage: false,
    }

    const component = renderer.create(
      <DataTable {...data} />,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // Default Props - Missing Cases
  it.each([
    {}, // empty
    { data: [] }, // missing isPercentage
    { isPercentage: false }, // missing data
  ])('Checking failing case for %p', (props) => {
    const component = renderer.create(
      <DataTable {...props} />,
    )
    const componentInstance = component.root
    expect(componentInstance.props).not.toEqual(props)
  })

  // Default Props - Valid Cases
  it.each([
    { data: [], isPercentage: true }, // data: checking empty array, isPercentage: non-default prop value
    { data: [], isPercentage: false }, // data: checking empty array, isPercentage: default prop value
    { data: [{}], isPercentage: false }, // data: checking array of obj
  ])('Checking passing case for %p', (props) => {
    const component = renderer.create(
      <DataTable {...props} />,
    )
    const componentInstance = component.root
    expect(componentInstance.props).toStrictEqual(props)
  })
})
