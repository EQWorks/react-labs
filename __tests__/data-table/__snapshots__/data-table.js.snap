// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataTable/> should render successfully 1`] = `
<div
  className="MuiTableContainer-root makeStyles-root-1"
>
  <table
    className="MuiTable-root MuiTable-stickyHeader"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      />
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    />
  </table>
</div>
`;
